va/caa.go

```diff
@@ -145,31 +133,38 @@ func parent(fqdn string) string {
 // conservative: We always do full tree-climbing on the original FQDN (by virtue
 // of parallelCAALookup. When the LegacyCAA flag is enabled, we also
 // do linear tree climbing on single-level aliases.
-func (va *ValidationAuthorityImpl) treeClimbingLookupCAA(ctx context.Context, fqdn string) ([]*dns.CAA, []*dns.CNAME, error) {
-	target := fqdn
-	// Limit CNAME chasing to break CNAME loops
-	for i := 0; i < 8 && target != ""; i++ {
-		caas, cnames, err := va.dnsClient.LookupCAA(ctx, target)
-		if err != nil {
-			return nil, nil, err
-		} else if len(caas) > 0 {
-			return caas, nil, nil
-		} else if len(cnames) > 0 {
-			target = findAlias(target, cnames)
-		} else if i == 0 {
-			// Special case: If there are no CNAMEs or CAAs on the very first lookup,
-			// skip the tree-climbing, since that's already implemented by parallelCAALookup.
-			// This minimizes duplicate lookups.
-			return nil, nil, nil
-		} else {
-			// If we're not on the first iteration (i.e. we've found a CNAME), do tree climbing.
-			target = parent(target)
+func (va *ValidationAuthorityImpl) treeClimbingLookupCAA(ctx context.Context, fqdn string) ([]*dns.CAA, error) {
+	// We will do a maximum of 8 tree-climbing queries to avoid CNAME/CAA hybrid
+	// loops
+	maxAttempts := 8
+	return va.treeClimbingLookupCAAWithCount(ctx, fqdn, &maxAttempts)
+}
+
+func (va *ValidationAuthorityImpl) treeClimbingLookupCAAWithCount(ctx context.Context, fqdn string, maxAttempts *int) ([]*dns.CAA, error) {
```

Comment:
`maxAttempts` might be better described as `attemptsRemaining` in the context of the loop w/ decrement

Reply:
Done.